### PR TITLE
Add a class for assist efficient traversal of dimensions in an Indexe…

### PR DESCRIPTION
**Pull request by baldersheim**
…dTensor.

@bratseth @jobergum PR

### CODE REVIEW

**Comment by baldersheim** [comment]
Now runtime of unit test is down from around 40s to below 8s.

File: vespajlib/src/main/java/com/yahoo/tensor/DirectIndexedAddress.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public void setIndex(int dimension, int index) {
+        if (index < 0 || index >= sizes.size(dimension)) {
+            throw new IndexOutOfBoundsException("Index " + index + " outside of [0," + sizes.size(dimension) + ">");
+        }
+        int diff = index - indexes[dimension];
+        directIndex += getStride(dimension) * diff;
+        indexes[dimension] = index;
+    }
+    /** Retrieve the index that can be used for direct lookup in an indexed tensor. */
+    public long getIndex() { return directIndex; }
```
Comment (bratseth) [inline vespajlib/src/main/java/com/yahoo/tensor/DirectIndexedAddress.java:33]: getDirectIndex?

**Review by bratseth — DISMISSED** [review]
Nice!

File: vespajlib/src/main/java/com/yahoo/tensor/DirectIndexedAddress.java (hunk 1 above)
Comment (baldersheim) [reply vespajlib/src/main/java/com/yahoo/tensor/DirectIndexedAddress.java]: Fixed

**Pull request merged**